.github/workflows/build.yml

```
@@ -270,17 +290,46 @@ jobs:
           rm -rf temp
           mv signed temp
 
-      - name: Push
+      - name: Generate VSA
+        if: env.shouldSign == 'true'
         env:
-          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
-          DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
+          AWS_KMS_REGION:  ${{ job.build.steps.auth.with.args.aws-region }}
+          AWS_KMS_KEY_ARN: ${{ job.build.steps.sign.env.AWS_KMS_KEY_ARN }}
         run: |
-          export DOCKER_CONFIG="$PWD/.docker"
-          mkdir "$DOCKER_CONFIG"
-          trap 'find "$DOCKER_CONFIG" -type f -exec shred -fuvz "{}" + || :; rm -rf "$DOCKER_CONFIG"' EXIT
-          docker login --username "$DOCKER_HUB_USERNAME" --password-stdin <<<"$DOCKER_HUB_PASSWORD"
-          unset DOCKER_HUB_USERNAME DOCKER_HUB_PASSWORD
+          args+=(
+            --mount "type=bind,src=$PWD/signed,dst=/doi-build/signed"
+
+            # https://explore.ggcr.dev/?repo=docker/image-signer-verifier
+            "$BASHBREW_IMAGE_SIGNER"
+
+            # verify is called here to generate a VSA (https://slsa.dev/spec/v1.0/verification_summary)
+            verify
+
+            -D # debug output
+
+            --platform="$BASHBREW_ARCH"
+            --vsa=true # whether or not to generate a VSA from verification output
 
+            --tuf-source=oci  # how to pull TUF data for policy and key material (oci or http)
+            --tuf-meta-source="$BASHBREW_TUF_META"  # TUF metadata URI
+            --tuf-targets-source="$BASHBREW_TUF_TARGETS"  # TUF targets URI
+
+            --policy-id="$BASHBREW_SIGNING_POLICY"  # policy stored in TUF for sbom/provenance verification
+
```

we can now set the verified levels and verifier via args instead of hardcoded in policy, eventually we will probably want to move the `verification-levels` back to policy and lookup the builder's verified SLSA level from a map of `builder.id` -> `SLSA level` and set that automatically.
```suggestion
--verified-levels="SLSA_BUILD_LEVEL_2"
--verifier-id="https://docker.io/verifier/docker-official-images"

```
`verifier-id` should be a [TypeURI](https://github.com/in-toto/attestation/blob/7aefca35a0f74a6e0cb397a8c4a76558f54de571/spec/v1/field_types.md#TypeURI). I made this one up just now.